### PR TITLE
Move retention policy to instance config

### PR DESCRIPTION
**Pull request by danvixent**

### CODE REVIEW

File: api/migrations/v20240517/projects.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
+package v20240517
```
Comment (subomi) [inline api/migrations/v20240517/projects.go:1]: @danvixent, we don't need to make this backwards compatible.

Comment (danvixent) [reply api/migrations/v20240517/projects.go:1]: @subomi okay!

File: cmd/hooks/hooks.go [hunk 2]
```diff
@@ -175,14 +177,16 @@ func PreRun(app *cli.App, db *postgres.Postgres) func(cmd *cobra.Command, args [
...
 		// update config singleton with the instance id
-		configRepo := postgres.NewConfigRepo(app.DB)
-		instCfg, err := configRepo.LoadConfiguration(cmd.Context())
-		if err != nil {
-			log.WithError(err).Error("Failed to load configuration")
-		} else {
-			cfg.InstanceId = instCfg.UID
-			if err = config.Override(&cfg); err != nil {
-				return err
+		if cmd.Use != "bootstrap" && cmd.Use != "version" { // bootstrap & version don't need this
```
Comment (jirevwe) [inline cmd/hooks/hooks.go:180]: Hmm, nice. I think `migrate` doesn't need this too. Can you add it?

File: config/config.go [hunk 3]
```diff
@@ -251,6 +255,11 @@ type SentryConfiguration struct {
 	DSN string `json:"dsn" envconfig:"CONVOY_SENTRY_DSN"`
 }
 
+type RetentionPolicyConfiguration struct {
```
Comment (jirevwe) [inline config/config.go:258]: Didn't we have this before? Or was it only in the project settings?

File: internal/pkg/exporter/exporter.go [hunk 4]
```diff
@@ -232,9 +234,27 @@ func getOutputWriter(out string) (io.WriteCloser, error) {
...
+//}
+//
+//func (c compressWriter) Close() error {
+//	if err := c.gw.Close(); err != nil {
+//		return err
+//	}
+//
+//	return c.file.Close()
+//}
+
```
Comment (jirevwe) [inline internal/pkg/exporter/exporter.go:257]: Shouldn't we remove this for now?

Comment (danvixent) [reply internal/pkg/exporter/exporter.go:257]: I figure we will need it in the future

File: config/config.go (hunk 3 above)
Comment (danvixent) [reply config/config.go:258]: Only project settings

Comment (jirevwe) [reply config/config.go:258]: ok

File: cmd/hooks/hooks.go (hunk 2 above)
Comment (danvixent) [reply cmd/hooks/hooks.go]: Done

**Pull request merged**